Pi+/Coxeter/Parametrized/FinLehmerEquiv.agda

```diff
@@ -0,0 +1,172 @@
+{-# OPTIONS --without-K --rewriting #-}
+
+module Pi+.Coxeter.Parametrized.FinLehmerEquiv where
+
+open import lib.Base
+open import lib.NType
+open import lib.NType2
+open import lib.Equivalence
+open import lib.Univalence
+open import lib.PathGroupoid
+open import lib.PathOver
+open import lib.types.Fin
+open import lib.types.List
+open import lib.types.Sigma
+open import lib.types.Coproduct
+open import lib.types.Unit
+open import lib.types.Nat using (<-has-all-paths; ltS)
+
+open import Pi+.Coxeter.Common.Lehmer
+open import Pi+.Coxeter.Common.InequalityEquiv
+open import Pi+.Coxeter.Common.Arithmetic
+
+open import Pi+.UFin.Monoidal
+open import Pi+.Extra
+open import Pi+.Misc
+
+≤→< : {k n : ℕ} -> (k ≤ n) -> (k < S n)
+≤→< z≤n = s≤s z≤n
+≤→< (s≤s p) = s≤s (≤→< p)
+
+LehmerInd : {n : ℕ} -> (Fin (S (S n))) × Lehmer n ≃ Lehmer (S n)
+LehmerInd {n} = equiv f g f-g g-f
+    where
+    f : _
+    f ((x , p) , l) = CanS l (≤-down2 ((–> <N≃< p)))
+    g : _
+    g (CanS l {r} p) = (r , <– <N≃< (≤→< p)) , l
+    f-g : _
+    f-g (CanS l {r} p) = ap (CanS l) (≤-has-all-paths _ _)
+    g-f : _
+    g-f ((x , p) , l) = pair= (pair= idp (<-has-all-paths _ _)) (↓-cst-in idp)
```

Comment:
Hm, this would be definitional if Lehmer was defined just by equations like factorial, instead of a type family. Do we rely on the type family definition elsewhere?